src/main/java/com/github/teamfusion/spyglassplus/common/entity/SpyglassStandEntity.java

```diff
@@ -148,11 +145,17 @@ protected void defineSynchedData() {
 	@Override
 	protected void readAdditionalSaveData(CompoundTag p_20052_) {
 		this.setHigh(p_20052_.getBoolean("IsHigh"));
+		ItemStack itemstack = this.getSpyGlass();
+		if (!itemstack.isEmpty()) {
+			p_20052_.put("Spyglass", itemstack.save(new CompoundTag()));
+		}
 	}
 
 	@Override
 	protected void addAdditionalSaveData(CompoundTag p_20139_) {
 		p_20139_.putBoolean("IsHigh", this.isHigh());
+		ItemStack itemstack = ItemStack.of(p_20139_.getCompound("Spyglass"));
+		this.setSpyGlass(itemstack);
```

Comment:
It looks like you mixed up the reading and adding for the ItemStack here. You're adding the ItemStack to the tag in the "read" method and reading the ItemStack from the tag in the "add" method.